src/MongoDB.Bson/IO/IBsonReaderExtensions.cs

```diff
@@ -88,6 +88,34 @@ public static BsonBinaryData ReadBinaryData(this IBsonReader reader, string name
             return reader.ReadBinaryData();
         }
 
+        /// <summary>
+        /// Reads a BSON binary data element from the reader temporarily setting the GuidRepresentation to Unspecified.
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <returns></returns>
+        [Obsolete("In V3 mode use ReadBinaryData instead.")]
+        public static BsonBinaryData ReadBinaryDataWithGuidRepresentationUnspecified(this IBsonReader reader)
```

Comment:
I still want to add this method. It's a little different than the previous method because it's now an extension method and based on PushSettings.

This method is called from two places and it's good to encapsulate the logic of temporarily overriding the GuidRepresentation in the settings.

I'm also fine with it being public. Anyone writing a custom serializer might very well need to call this method.

Reply:
Ack